Understat match stats and players

jruizcabrejos opened the pull request:
Currently, understat_match data only retrieves shots data.

This 2 functions should retrieve both the Stats table and the player data for each match.

tonyelhabr commented on R/understat_match_players.R:
```diff
@@ -0,0 +1,53 @@
+#'
+#' @export
+
+understat_match_players <- function(match_url) {
```
Please indent the body of this function (add 2 spaces at the beginning of each line). Technically it's not wrong R syntax and we are pretty relaxed with spacing/indentation rules, but that is fairly unconventional.

tonyelhabr commented on R/understat_match_players.R:
```diff
@@ -0,0 +1,53 @@
+match_player_data_home <- do.call(rbind.data.frame, match_player_data$h)
+match_player_data_away <- do.call(rbind.data.frame, match_player_data$a)
+
+match_player_data <- bind_rows(match_player_data_home,match_player_data_away) %>%
```
Please add namespace prefixing (e.g. `dplyr::`) here and below.

tonyelhabr commented on R/understat_match_players.R:
```diff
@@ -0,0 +1,53 @@
+match_player_data <- bind_rows(match_player_data_home,match_player_data_away) %>%
+  mutate(match_id = match_id)  %>%
+
+  select(match_id,  team_id,
+         team_status = h_a,
+         player_id, swap_id = id,
+         player, position, positionOrder,
+         time_played = time,
+         everything()) %>%
```
Please type out all of the column names and coerce numeric values to numerics (or integers) where it makes sense to do so, e.g. `time_played`, `goals`, etc.

We don't like using `everything()` because it makes it hard to find-and-replace things for refactoring, and makes the code susceptible to source data changes (i.e. changes in column existence from understat). It is fine for non-package code.

tonyelhabr commented on R/understat_match_stats.R:
```diff
@@ -0,0 +1,56 @@
+
+    home_team = away[1],
+    home_chances = away[2],
+    home_goals = home[3],
+    home_xG = home[4],
+    home_shots = home[5],
+    home_shot_on_target = home[6],
+    home_deep = home[7],
+    home_PPDA = home[8],
+    home_xPTS = home[9],
```
please convert to numeric/integer, i.e. `as.numeric` for `home_xG`, etc., and `as.integer` for `home_goals, `home_shots`, etc.

and please doe the same for the away columns

tonyelhabr reviewed (commented):
This looks ok to me, and I'll be happy to approve when my feedback is addressed. Oh, and one other thing not mentioned in in-line comments--please add tests for each function in `tests/testthat/test-understat.R`. 

This is a great first pass! I'm really happy to see your contribution.

jruizcabrejos commented:
Great! I have addressed all the comments in the review (namespaces, indentation, and variable coercion). Added the tests for both functions and updated some of the documentation.

For `understat_match_stats` it is further explained a limitation for the variable "chances". For example, in the next screenshot [(match 22275)](https://understat.com/match/22275) the 2% is being retrieved as an empty character `""`.  These empty values are being set to `NA`.

The 90% does return correctly (and it is being processed into a `0.9` in the final output)

![image](https://github.com/JaseZiv/worldfootballR/assets/25235690/e9372930-22b5-43fd-9ea5-6bf0e1dceafe)

tonyelhabr commented:
i just pushed some final touch-ups. testing in Github Actions now. some transfermarket tests may fail, but we should see the understat tests pass

tonyelhabr commented on R/understat_match_players.R:
```diff
@@ -0,0 +1,74 @@
+#'
+#' @export
+
+understat_match_players <- function(match_url) {
```
my only new feedback on this is that there might have been a cleaner way of doing all of this. i noticed that [`{understatr}` seems to have a more straightforward approach](https://github.com/ewenme/understatr/blob/dd6e48fb3b86174b9346e9b08a821045c9415d4d/R/matches.R), although i haven't thoroughly checked the details. it could be the same exact implementation.

tonyelhabr replied:
but i don't think this function should be changed at this point

JaseZiv commented:
Thanks for the contribution @jruizcabrejos!

Thanks also for the work reviewing @tonyelhabr!
